### PR TITLE
Add ACLE legacy releases

### PR DESCRIPTION
**Pull request by vhscampos**
---
name: Pull request
about: Technical issues, document format problems, bugs in scripts or feature proposal.

---

**Thank you for submitting a pull request!**

If this PR is about a bugfix:

Please use the bugfix label and make sure to go through the checklist below.

If this PR is about a proposal:

We are looking forward to evaluate your proposal, and if possible to
make it part of the Arm C Language Extension (ACLE) specifications.

We would like to encourage you reading through the [contribution
guidelines](https://github.com/ARM-software/acle/blob/main/CONTRIBUTING.md), in particular the section on [submitting
a proposal](https://github.com/ARM-software/acle/blob/main/CONTRIBUTING.md#proposals-for-new-content).

Please use the proposal label.

As for any pull request, please make sure to go through the below
checklist.

Checklist: (mark with ``X`` those which apply)

* [x] If an issue reporting the bug exists, I have mentioned it in the
      PR (do not bother creating the issue if all you want to do is
      fixing the bug yourself).
* [ ] I have added/updated the `SPDX-FileCopyrightText` lines on top
      of any file I have edited. Format is `SPDX-FileCopyrightText:
      Copyright {year} {entity or name} <{contact informations}>`
      (Please update existing copyright lines if applicable. You can
      specify year ranges with hyphen , as in `2017-2019`, and use
      commas to separate gaps, as in `2018-2020, 2022`).
* [x] I have updated the `Copyright` section of the sources of the
      specification I have edited (this will show up in the text
      rendered in the PDF and other output format supported). The
      format is the same described in the previous item.
* [x] I have run the CI scripts (if applicable, as they might be
      tricky to set up on non-*nix machines). The sequence can be
      found in the [contribution
      guidelines](https://github.com/ARM-software/acle/blob/main/CONTRIBUTING.md#continuous-integration). Don't
      worry if you cannot run these scripts on your machine, your
      patch will be automatically checked in the Actions of the pull
      request.
* [ ] I have added an item that describes the changes I have
      introduced in this PR in the section **Changes for next
      release** of the section **Change Control**/**Document history**
      of the document. Create **Changes for next release** if it does
      not exist. Notice that changes that are not modifying the
      content and rendering of the specifications (both HTML and PDF)
      do not need to be listed.
* [x] When modifying content and/or its rendering, I have checked the
      correctness of the result in the PDF output (please refer to the
      instructions on [how to build the PDFs
      locally](https://github.com/ARM-software/acle/blob/main/CONTRIBUTING.md#continuous-integration)).
* [x] The variable `draftversion` is set to `true` in the YAML header
      of the sources of the specifications I have modified.
* [ ] Please *DO NOT* add my GitHub profile to the list of contributors
      in the [README](https://github.com/ARM-software/acle/blob/main/README.md#contributors-) page of the project.

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
-<!-- SPDX-FileCopyrightText: Copyright 2021 Arm Limited and/or its affiliates <open-source-office@arm.com> -->
+<!-- SPDX-FileCopyrightText: Copyright 2021, 2023 Arm Limited and/or its affiliates <open-source-office@arm.com> -->
 <!-- SPDX-License-Identifier: Apache-2.0 -->
 
 # License for the specifications
 
-All the ACLE documents themselves are not dependent on any assets
+All the current ACLE documents themselves are not dependent on any assets
```
Comment (stuij) [inline LICENSE.md:6]: When I read `current` without any other context I wonder what current means exactly. When is a document current.

File: LICENSE.md [hunk 2]
```diff
@@ -18,6 +18,13 @@ relative subfolders:
 * [mve_intrinsics](mve_intrinsics)
 * [neon_intrinsics](neon_intrinsics)
 
+# License for the legacy releases
+
+All the legacy ACLE releases are licensed under an Arm proprietary license.
+They predate the new license of the current versions of the documents.
```
Comment (stuij) [inline LICENSE.md:24]: as opposed to the `current` here which frames the current versions against the proprietary licensed documents

File: legacyreleases/LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<!-- SPDX-FileCopyrightText: Copyright 2023 Arm Limited and/or its affiliates <open-source-office@arm.com> -->
+<!-- SPDX-License-Identifier: CC-BY-SA-4.0 AND Apache-Patent-License -->
+
+# License
```
Comment (stuij) [inline legacyreleases/LICENSE.md:4]: I wonder if this does more harm than good. None of the documents are licensed under Creative Commons, and I don't think we need a Creative Commons license for the readme.

**Review by stuij — COMMENTED** [review]
Yay, legacy ACLE documents united with their open source counterparts!

File: LICENSE.md (hunk 1 above)
Comment (vhscampos) [reply LICENSE.md]: Fixed. "current" -> "open-source", and made it clearer that the legacy ones are closed-source.

File: LICENSE.md (hunk 2 above)
Comment (vhscampos) [reply LICENSE.md]: Fixed as already mentioned.

File: legacyreleases/LICENSE.md (hunk 3 above)
Comment (vhscampos) [reply legacyreleases/LICENSE.md]: In the legacyreleases folder I left only the README file, which is itself under Apache 2.0. So no mention of CC anymore under this folder.

**Review by stuij — APPROVED** [review]
LGTM

**Comment by vhscampos** [comment]
@all-contributors Please add @vhscampos for code.

**Pull request merged**